### PR TITLE
Spring cleaning for the TPR parser

### PR DESCRIPTION
**Pull request by jbarnoud**
Some cleaning of the TPR parser.

* [x] Remove code that dealt with python 2 compatibility
* [x] Use f-strings instead of format
* [x] Remove dead code
* [x] Increase test coverage
* [ ] Add type annotations

PR Checklist
------------
 - [x] Tests?
 - ~[ ] Docs?~
 - [x] CHANGELOG updated?
 - ~[ ] Issue raised/referenced?~

### CODE REVIEW

File: testsuite/MDAnalysisTests/data/tprs/virtual_sites/dummy.top [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+; name mass charge ptype c6          c12
+TA     72.0 0.000  A     0.24145E-00 0.26027E-02
+
+[ moleculetype ]
+; molname  	nrexcl
+TEST 	    	1
+
+[ atoms ]
+;id 	type 	resnr 	residu 	atom 	cgnr 	charge   mass
+1       TA      1       T       T1      1       0
```
Comment (orbeckst) [inline testsuite/MDAnalysisTests/data/tprs/virtual_sites/dummy.top:19]: I am surprised that you didn't name the atoms with emojis...

Comment (jbarnoud) [reply testsuite/MDAnalysisTests/data/tprs/virtual_sites/dummy.top:19]: It i a missed opportunity! Though, emojis are not officially supported by gromacs, so maybe they are a bit too hardcore for the tests :stuck_out_tongue:

**Comment by richardjgowers** [comment]
@jbarnoud we can merge this as-is and you can do tests/type annotations in a future PR

**Comment by jbarnoud** [comment]
@richardjgowers I am done with the tests. Assuming the tests pass, it is good for review. I can add type annotations in a later PR.

**Pull request merged**